### ISSUE TITLE
Saveloading error on emp04 mission specific units again

### ISSUE THREAD

**Issue by tothxa** [widelands/widelands#5483]
**Describe the bug**
v1.0 savegames of emp04 fail to load with latest master

**To Reproduce**
1. Start a game of emp04 in v1.0
2. Save it: [emp04_v1.0_wgf.zip](https://github.com/widelands/widelands/files/8628213/emp04_v1.0_wgf.zip)
3. Load the saved game in current development version.

**Expected behavior**
Game should be loaded and be playable.

**Version:**
a7704bd / first noticed in flatpak beta of 99ccb31.
Worked with a83c094.

**Additional context**
stdout:
```
[00:00:01.419 real] WARNING: Unused key "empire" in LuaTable. Please report as a bug.
[00:00:02.707 real] INFO: Songset: Loaded song "music/menu.ogg"
[00:00:04.186 real] ERROR: Widelands could not load the file "/tmp/emp04_test.wgf". The file format seems to be incompatible.
[00:00:04.186 real] ERROR: Error message:
[00:00:04.186 real] ERROR: buildings: tribe empire does not define building type "empire_foresters_house1"
```

#5480 does NOT fix it.

**Comment by Noordfrees** [comment]
Not *again*! ;)

**Comment by Noordfrees** [comment]
I added a fix to #5480 in the hope it won't cause *other* regressions ;)
